### PR TITLE
[azservicebus/azeventhubs] Enable custom endpoints, which allows our library to be used with TCP proxies

### PR DESCRIPTION
**Pull request by richardpark-msft**
This is a feature that .NET has, and it's handy for when you need to route your TCP traffic through a proxy. It also enables some testing scenarios.

APIViews:
- azservicebus: https://apiview.dev/Assemblies/Review/0579bbe5a2324f17a48f1a2bca2ff383?revisionId=e548a02147214107802f26ba47e2a9bf&diffOnly=True&doc=False&diffRevisionId=868bd2a67e9641a3bad83fb66cde2dc4
- azeventhubs: https://apiview.dev/Assemblies/Review/e1445c654ad0416c821be30b02dc445c/3c935256189a453ead108bc9fd3ec9f2?diffRevisionId=51318a1ab3b44f2486db3d3f860834f1&doc=False&diffOnly=True

### CODE REVIEW

**Review by JeffreyRichter — APPROVED** [review]

File: sdk/messaging/azeventhubs/consumer_client.go [hunk 1]
```diff
@@ -20,6 +20,9 @@ type ConsumerClientOptions struct {
 	// ApplicationID is used as the identifier when setting the User-Agent property.
 	ApplicationID string
 
+	// A custom endpoint address that can be used when establishing the connection to the service.
+	CustomEndpoint string
```
Comment (RickWinter) [inline sdk/messaging/azeventhubs/consumer_client.go:24]: Is this name consistent with other languages?

Comment (richardpark-msft) [reply sdk/messaging/azeventhubs/consumer_client.go:24]: It's shorter - the other languages have gone with `CustomEndpointAddress` but our SDKs tend to just use 'endpoint' as the terminology for the same thing.

Comment (richardpark-msft) [reply sdk/messaging/azeventhubs/consumer_client.go:24]: (talked offline with Rick, I'm going to leave this one as-is as)

**Pull request merged**